bundles/org.openhab.binding.tibber/src/main/java/org/openhab/binding/tibber/internal/handler/TibberHandler.java

```diff
@@ -0,0 +1,471 @@
+                }
+                if (myObject.has("minPower")) {
+                    updateChannel(LIVE_MINPOWER, myObject.get("minPower").toString());
+                }
+                if (myObject.has("averagePower")) {
+                    updateChannel(LIVE_AVERAGEPOWER, myObject.get("averagePower").toString());
+                }
+                if (myObject.has("maxPower")) {
+                    updateChannel(LIVE_MAXPOWER, myObject.get("maxPower").toString());
+                }
+                if (myObject.has("voltagePhase1")) {
+                    updateChannel(LIVE_VOLTAGE1, myObject.get("voltagePhase1").toString());
+                }
+                if (myObject.has("voltagePhase2")) {
+                    updateChannel(LIVE_VOLTAGE2, myObject.get("voltagePhase2").toString());
+                }
+                if (myObject.has("voltagePhase3")) {
+                    updateChannel(LIVE_VOLTAGE3, myObject.get("voltagePhase3").toString());
+                }
+                if (myObject.has("currentPhase1")) {
+                    updateChannel(LIVE_CURRENT1, myObject.get("currentPhase1").toString());
+                }
+                if (myObject.has("currentPhase2")) {
+                    updateChannel(LIVE_CURRENT2, myObject.get("currentPhase2").toString());
+                }
+                if (myObject.has("currentPhase3")) {
+                    updateChannel(LIVE_CURRENT3, myObject.get("currentPhase3").toString());
+                }
+                if (myObject.has("powerProduction")) {
+                    updateChannel(LIVE_POWERPRODUCTION, myObject.get("powerProduction").toString());
+                }
+                if (myObject.has("accumulatedProduction")) {
+                    updateChannel(LIVE_ACCUMULATEDPRODUCTION, myObject.get("accumulatedProduction").toString());
+                }
+                if (myObject.has("minPowerProduction")) {
+                    updateChannel(LIVE_MINPOWERPRODUCTION, myObject.get("minPowerProduction").toString());
+                }
+                if (myObject.has("maxPowerProduction")) {
+                    updateChannel(LIVE_MAXPOWERPRODUCTION, myObject.get("maxPowerProduction").toString());
+                }
+            } else {
+                logger.debug("Unknown live response from Tibber");
+            }
+        }
+
+        private void sendMessage(String message) throws IOException {
+            logger.debug("Send message: {}", message);
+            Session session = TibberHandler.this.session;
+            if (session != null) {
+                session.getRemote().sendString(message);
+            }
+        }
+
+        public void onWebSocketBinary(byte[] payload, int offset, int len) {
+            logger.debug("WebSocketBinary({}, {}, '{}')", offset, len, Arrays.toString(payload));
+        }
+
+        public void onWebSocketText(String message) {
+            logger.debug("WebSocketText('{}')", message);
```

Comment:
So you only use websocket to send messages? Why do you need a websocket if you aren't doing two way communication?

Reply:
Unsure what you are pointing to here.

According to vendor specific service/API, one need to establish a websocket connection and send an initialise message to start the stream from the service/API.

And that is what the binding achieve as is: An initialise message is sent, and a stream of data is received every 2 seconds.

Reply:
I was only pointing out that this method doesn't do anything, this is the method that would handle inbound messages from the websocket correct?

Reply:
The websocket methods are just based on default jetty library. Although, there might seem like `public void onMessage(String message)` and `public void onWebSocketText(String message)` would handle the same thing, so I have removed the unused methods.